bfss3/bfss3.go

```diff
@@ -453,7 +464,7 @@ func (i *iterator) fetchNextPage() error {
 	i.page = i.page[:0]
 	i.pos = -1
 
-	res, err := i.parent.ListObjectsV2WithContext(i.ctx, &s3.ListObjectsV2Input{
+	res, err := i.parent.ListObjectsV2(i.ctx, &s3.ListObjectsV2Input{
```

Comment:
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/s3/#ListObjectsV2Paginator could be used in future, wanted to keep diff small for now. Tbh, I don't think that migrating to Paginator is worth the effort - I'm not even sure if it would save many LOCs to justify bothering.

Reply:
I prefer to use those helpers as they keep improving and optimising them (including new API features/headers etc). But I agree, you can/should do that in a separate PR